gateway/dataimpl/server_v1/errorhandler.go

```diff
@@ -436,12 +436,20 @@ func (e ErrorHandler) NewSdPathTooBigStatus(baseErr error, sdPath string) *statu
 }
 
 func (e ErrorHandler) NewSdBadValueStatus(baseErr error, sdPath string) *status.Status {
-	st := status.New(codes.FailedPrecondition,
+	st := status.New(codes.InvalidArgument,
 		fmt.Sprintf("Subdocument operation content for path '%s' would invalidate the JSON if added to the document.",
 			sdPath))
+	st = e.tryAttachExtraContext(st, baseErr)
+	return st
+}
+
+func (e ErrorHandler) NewSdValueOutOfRangeStatus(baseErr error, sdPath string) *status.Status {
+	st := status.New(codes.FailedPrecondition,
+		fmt.Sprintf("Counter operation content for path '%s' would put the JSON value out of range.",
+			sdPath))
 	st = e.tryAttachStatusDetails(st, &epb.PreconditionFailure{
 		Violations: []*epb.PreconditionFailure_Violation{{
-			Type:        "WOULD_INVALIDATE_JSON",
+			Type:        "VALUE_OUT_OF_RANGE",
```

Comment:
N.b. this is a breaking change.  We can accommodate it as long as this is merged in the next couple of days, but after we're GA changes like this will be problematic.

It worries me that, while we have mechanisms to let the SDK opt-in to new APIs (the versioning), we don't have a mechanism to allow the SDK to opt-in to server changes like this.  Maybe we can open a separate discussion on that?

Reply:
@programmatix I agree and already signalled to Ray that we should discuss this soon: https://couchbase.slack.com/archives/C04DB7P157T/p1698695110596189